include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -49,11 +49,29 @@
 #    define _ONEDPL_PREDEFINED_POLICIES 1
 #endif
 
-#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND)
+// Check availability of parallel backends
+#if __has_include(<tbb/tbb.h>)
+#    define _ONEDPL_TBB_AVAILABLE 1
+#endif
+#if ONEDPL_USE_TBB_BACKEND && !_ONEDPL_TBB_AVAILABLE
+#    error "TBB backend is selected, but TBB is not found"
+#endif
+// Check OpenMP availability only during host code compilation
+#if !defined(__SYCL_DEVICE_ONLY__)
+#    if defined(_OPENMP)
+#        define _ONEDPL_OPENMP_AVAILABLE 1
+#    endif
+#    if ONEDPL_USE_OPENMP_BACKEND && !_ONEDPL_OPENMP_AVAILABLE
+#        error "OpenMP backend is selected, but OpenMP is not found"
+#    endif
+#endif //no __SYCL_DEVICE_ONLY__
+
+// Select a parallel backend
+#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND && _ONEDPL_TBB_AVAILABLE)
 #    define _ONEDPL_PAR_BACKEND_TBB 1
 #endif
 
-#if ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && defined(_OPENMP))
+#if ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && _ONEDPL_OPENMP_AVAILABLE)
```

Comment:
Let's move the logic in the lines 69-80 to the place where it is used, `parallel_backend.h`. This logic is very confusing and hard to understand without knowing how it is used there.